### ISSUE TITLE
Is scopt composable?

### ISSUE THREAD

**Issue by frgomes** [scopt/scopt#215]
I would like to define separate parsers responsible for disjoint sets of options and compose them as needed, for example: I would define one parser per submodule of my project. When I define an application, I define which modules are needed and, doing so, the full command line parser would be made of the composition of the disjoint command line parsers defined in the modules involved. Is that scenario possible with scopt? How could that be achieved?

**Comment by eed3si9n** [comment]
I have a local branch I used to work on that makes scopt parsers monadic - https://github.com/scopt/scopt/compare/scopt3...eed3si9n:topic/monadic
This would allow composition like this:

```scala
  lazy val groupParser1: Parser[Unit, Config] = {
    import builder._
    for {
      _ <- programName("scopt")
      _ <- head("scopt", "3.x")
      _ <- opt[Unit]('a', "alice")
      _ <- opt[Unit]('b', "bob")
      _ <- opt[Unit]("alicebob").abbr("ab").action( (x, c) => c.copy(flag = true) )
      // _ <_ help("help")
    } yield ()
  }
```

**Comment by frgomes** [comment]
That looks nice! Any idea (if and) when it will be merged?

**Comment by eed3si9n** [comment]
Released scopt 4.0.0-RC2. See http://eed3si9n.com/scopt4 for details.
